docs/user/troubleshootingbootstrap.md

```diff
@@ -0,0 +1,220 @@
+* `kubelet.log`, the kubelet service is responsible for running the kubelet on the bootstrap host. The kubelet on the bootstrap host is responsible for running the static pods for etcd, bootstrap-kube-controlplane and various other operators in bootstrap mode.
+* `approve-csr.log`, the approve-csr unit is responsible for allowing control-plane machines to join OpenShift cluster. This unit performs the job of in-cluster approver while the bootstrapping is in progress.
+
+* `bootkube.log`, the bootkube service is the unit that performs the bootstrapping of OpenShift clusters using all the operators. This service is respnsible for running all the required steps to bootstrap the API and then wait for success.
+
+There might also be other services that are important for some platforms like openstack, that will have logs in this directory.
+
+#### directory: bootstrap/pods
+
+The pods directory contains the information and logs from all the render commands for various operators run by the bootkube unit.
+
+For each container the directory has two files,
+
+* `<human readable id>.log`, which contains the log of the container.
+* `<human readable id>.inspect`, which containts the information about the container like the image, volume mounts, arguments etc.
+
+### directory: resources
+
+The resources direcory contains various Kubernetes objects that are present in the cluster. These resources are pulled using the bootstrap API running on the bootstrap host.
+
+### directory: rendered-assets
+
+The rendered-assets directory contains all the files and directories created by the bootkube unit using various render command for operators. This directory is a snapshot of the `/opt/openshift` directory on the bootstrap-host.
+
+### directory: control-plane
+
+The control-plane directory contains logs for each control-plane host. It contains a sub directory for each control-plane host, usually the IP address of the hosts.
+
+```console
+control-plane
+├── 10.0.128.114
+│   ├── containers
+│   ├── failed-units.txt
+│   ├── journals
+│   └── unit-status
+├── 10.0.142.138
+│   ├── containers
+│   ├── failed-units.txt
+│   ├── journals
+│   └── unit-status
+└── 10.0.148.48
+    ├── containers
+    ├── failed-units.txt
+    ├── journals
+    └── unit-status
+
+12 directories, 3 files
+```
+
+#### directory: control-plane/*/containers
+
+The containers directory contains the descriptions and logs from all the containers created by the kubelet using CRIO on the control-plane host. The files are same as [containers directory][#directory-bootstrap-containers] on bootstrap host.
+
+#### directory: control-plane/*/journals
+
+The journals directory contains the logs of **important** units on the control plane hosts. The list of such units is,
+
+* `crio.log`
+* `kubelet.log`
+* `machine-config-daemon-host.log` and `pivot.log`, these files have logs for RHCOS pivot related actions on the control plane host.
+
+## Common Failures
```

Comment:
I would still like the installer to grow diagnostics for common failures (#2569).  Any thoughts about whether we can get an up/down decision on that direction once 4.6 splits off from master?